appinfoflex.go

```diff
@@ -28,6 +28,14 @@ func InternalNewAppengineInfoFromContext(c context.Context) AppengineInfo {
 var NewAppengineInfoFromContext = InternalNewAppengineInfoFromContext
 
 func (ai AppengineInfoFlex) AppID() string {
+	if project := os.Getenv("GOOGLE_CLOUD_PROJECT_OVERRIDE"); project != "" {
+		return project
+	}
+
+	return os.Getenv("GOOGLE_CLOUD_PROJECT")
+}
+
+func (ai AppengineInfoFlex) AppIDHosted() string {
```

Comment:
Along the same lines as what andy said, I think cleaner function names would be good here.  From a dev perspective it is not clear which of these you should be using if you need to make a new call somewhere.  Good names are hard :( 

Something that conveys that one of these always returns the native project as in the project you're executing in.  The other one returns where your data is.